src/Console/Command/ExportProducts.php

```diff
@@ -0,0 +1,139 @@
+use Omikron\Factfinder\Model\Config\CommunicationConfig;
+
+class ExportProducts extends \Symfony\Component\Console\Command\Command
+{
+    /** @var ScopeConfigInterface */
+    private $scopeConfig;
+
+    /** @var StoreEmulation */
+    private $storeEmulation;
+
+    /** @var FeedGeneratorFactory */
+    private $feedGeneratorFactory;
+
+    /** @var StoreManagerInterface */
+    private $storeManager;
+
+    /** @var CommunicationConfig */
+    private $communicationConfig;
+
+    /** @var CsvFactory */
+    private $csvFactory;
+
+    /** @var FtpUploader */
+    private $ftpUploader;
+
+    /** @var string */
+    private $feedType;
+
+    /** @var PushImport */
+    private $pushImport;
+
+    /** @var State */
+    private $state;
+
+    /** @var Filesystem */
+    private $filesystem;
+
+    public function __construct(
+        ScopeConfigInterface $scopeConfig,
+        StoreManagerInterface $storeManager,
+        FeedGeneratorFactory $feedFactory,
+        StoreEmulation $emulation,
+        CsvFactory $csvFactory,
+        FtpUploader $ftpUploader,
+        CommunicationConfig $communicationConfig,
+        PushImport $pushImport,
+        State $state,
+        Filesystem $filesystem,
+        string $name = null)
+    {
+        $this->scopeConfig = $scopeConfig;
+        $this->storeManager = $storeManager;
+        $this->feedGeneratorFactory = $feedFactory;
+        $this->storeEmulation = $emulation;
+        $this->csvFactory = $csvFactory;
+        $this->ftpUploader = $ftpUploader;
+        $this->communicationConfig = $communicationConfig;
+        $this->pushImport = $pushImport;
+        $this->state = $state;
+        $this->filesystem = $filesystem;
+        parent::__construct($name);
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    protected function configure()
+    {
+        $this->setName('factfinder:export:products')->setDescription('Export Factfinder Product Data as CSV file');
+
+        $this->addOption('store', 's', InputOption::VALUE_OPTIONAL, 'Store ID or Store Code');
+        $this->addOption('skip-ftp-upload', null, InputOption::VALUE_OPTIONAL, 'Skip FTP Upload');
+        $this->addOption('skip-push-import', null, InputOption::VALUE_OPTIONAL, 'Skip Push Import');
+
+        parent::configure();
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    protected function execute(InputInterface $input, OutputInterface $output)
+    {
+        $this->state->setAreaCode('frontend');
+
+        if ($storeId = $input->getOption('store')) {
+            $storeIds = [$this->storeManager->getStore($storeId)->getId()];
+        } else {
+            $storeIds = array_map(
+                function ($store) {
+                    return $store->getId();
+                },
+                $this->storeManager->getStores()
+            );
+        }
+        foreach ($storeIds as $storeId) {
+            $this->storeEmulation->runInStore(
+                (int)$storeId,
```

Comment:
Please check against our coding standards, phpcbf can fix these warnings.

Reply:
I checked against magento-ecg/coding-standard/EcgM2, it passes that. Are there more coding standards which are not referenced in the composer.json?

Reply:
We include a phpcs.xml.dist, but it's not so bad :-)